mm2src/coins/solana.rs

```diff
@@ -377,6 +426,311 @@ impl SolanaCoin {
         let guard = self.spl_tokens_infos.lock().unwrap();
         (*guard).clone()
     }
+
+    fn send_hash_time_locked_payment(&self, args: SendPaymentArgs<'_>) -> SolTxFut {
+        let receiver = Pubkey::new(args.other_pubkey.iter().as_slice());
+        let swap_program_id = Pubkey::new(args.swap_contract_address.as_ref().unwrap().as_slice());
+        let amount = sol_to_lamports(args.amount.to_f64().unwrap());
+        let secret_hash: [u8; 32] = <[u8; 32]>::try_from(args.secret_hash).expect("unable to convert to 32 byte array");
+        let (vault_pda, vault_pda_data, vault_bump_seed, vault_bump_seed_data, rent_exemption_lamports) =
+            self.create_vaults(args.time_lock, secret_hash, swap_program_id, 41);
+        let swap_instruction = AtomicSwapInstruction::LamportsPayment {
+            secret_hash,
+            lock_time: args.time_lock,
+            amount,
+            receiver,
+            rent_exemption_lamports,
+            vault_bump_seed,
+            vault_bump_seed_data,
+        };
+
+        let accounts = vec![
+            AccountMeta::new(self.key_pair.pubkey(), true), // Marked as signer
+            AccountMeta::new(vault_pda_data, false),        // Not a signer
+            AccountMeta::new(vault_pda, false),             // Not a signer
+            AccountMeta::new(solana_program::system_program::id(), false), //system_program must be included
+        ];
+        self.sign_and_send_transaction(swap_program_id, accounts, swap_instruction.pack())
+    }
+
+    fn spend_hash_time_locked_payment(&self, args: SpendPaymentArgs) -> SolTxFut {
+        let sender = Pubkey::new(args.other_pubkey.iter().as_slice());
+        let swap_program_id = Pubkey::new(args.swap_contract_address.as_ref().unwrap().as_slice());
+        let secret: [u8; 32] = <[u8; 32]>::try_from(args.secret).unwrap();
+        let secret_hash = sha256(secret.as_slice());
+        let (lock_time, _secret_hash, amount, token_program) =
+            self.get_transaction_details(args.other_payment_tx).unwrap();
```

Comment:
it's would be good to validate that the secret hash from the transaction details is the same as the hash of the secret.

Reply:
done

Reply:
`_secret_hash` is still prefixed with `_` when it's now being used https://github.com/KomodoPlatform/komodo-defi-framework/blob/ac8cc07536751d8145949f0542380bcb768ba45c/mm2src/coins/solana.rs#L481